bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/component/Climate.java

```diff
@@ -267,11 +313,24 @@ ComponentChannelType.SWITCH, new OnOffValue(), updateListener, null,
                 channelConfiguration.modeCommandTemplate, channelConfiguration.modeCommandTopic,
                 channelConfiguration.modeStateTemplate, channelConfiguration.modeStateTopic, commandFilter);
 
+        buildOptionalChannel(PRESET_MODE_CH_ID, ComponentChannelType.STRING,
+                new TextValue(channelConfiguration.presetModes.toArray(new String[0])), updateListener,
+                channelConfiguration.presetModeCommandTemplate, channelConfiguration.presetModeCommandTopic,
+                channelConfiguration.presetModeStateTemplate, channelConfiguration.presetModeStateTopic, commandFilter);
+
         buildOptionalChannel(SWING_CH_ID, ComponentChannelType.STRING,
                 new TextValue(channelConfiguration.swingModes.toArray(new String[0])), updateListener,
                 channelConfiguration.swingCommandTemplate, channelConfiguration.swingCommandTopic,
                 channelConfiguration.swingStateTemplate, channelConfiguration.swingStateTopic, commandFilter);
 
+        // Probably there should be PercentageValue, but I'm not sure how to use it
```

Comment:
Regarding this comment, did you test it? To me this seems valid.

Reply:
I have tested it with NumberValue and it works. I was looking at PercentageValue for a while, but I was afraid not to do some bigger bug. Probably you expirieced coders will be able to change it, if you will, without much troubles.